zellij-utils/src/input/plugins.rs

```diff
@@ -0,0 +1,285 @@
+    fn default() -> Self {
+        Self::get_plugins_with_default(Plugins::new())
+    }
+}
+
+impl TryFrom<PluginsFromYaml> for Plugins {
+    type Error = PluginsError;
+
+    fn try_from(yaml: PluginsFromYaml) -> Result<Self, PluginsError> {
+        let mut plugins = HashMap::new();
+        for plugin in yaml.0 {
+            if plugins.contains_key(&plugin.tag) {
+                return Err(PluginsError::DuplicatePlugins(plugin.tag));
+            }
+            plugins.insert(plugin.tag.clone(), plugin.into());
+        }
+
+        Ok(Plugins(plugins))
+    }
+}
+
+impl From<PluginFromYaml> for Plugin {
+    fn from(plugin: PluginFromYaml) -> Self {
+        Plugin {
+            path: plugin.path,
+            tag: plugin.tag,
+            run: match plugin.run {
+                PluginTypeFromYaml::OncePerPane => PluginType::OncePerPane(None),
+                PluginTypeFromYaml::Headless => PluginType::Headless,
+            },
+            _allow_exec_host_cmd: plugin._allow_exec_host_cmd,
+        }
+    }
+}
+
+/// Plugin metadata
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct Plugin {
+    /// Path of the plugin, see resolve_wasm_bytes for resolution semantics
+    pub path: PathBuf,
+    /// Tag used to identify the plugin in layout and config yaml files
+    pub tag: PluginTag,
+    /// Plugin type
+    pub run: PluginType,
+    /// Allow command execution from plugin
+    pub _allow_exec_host_cmd: bool,
+}
+
+impl Plugin {
+    /// Resolve wasm plugin bytes for the plugin path and given plugin directory. Attempts to first
+    /// resolve the plugin path as an absolute path, then adds a ".wasm" extension to the path and
+    /// resolves that, finally we use the plugin directoy joined with the path with an appended
+    /// ".wasm" extension. So if our path is "tab-bar" and the given plugin dir is
+    /// "/home/bob/.zellij/plugins" the lookup chain will be this:
+    ///
+    /// ```bash
+    ///   /tab-bar
+    ///   /tab-bar.wasm
+    ///   /home/bob/.zellij/plugins/tab-bar.wasm
+    /// ```
+    ///
+    pub fn resolve_wasm_bytes(&self, plugin_dir: &Path) -> Option<Vec<u8>> {
+        fs::read(&self.path)
+            .or_else(|_| fs::read(&self.path.with_extension("wasm")))
+            .or_else(|_| fs::read(plugin_dir.join(&self.path).with_extension("wasm")))
+            .ok()
+    }
+
+    /// Sets the tab index inside of the plugin type of the run field.
+    pub fn set_tab_index(&mut self, tab_index: usize) {
+        match self.run {
+            PluginType::OncePerPane(..) => {
+                self.run = PluginType::OncePerPane(Some(tab_index));
+            }
+            PluginType::Headless => {}
+        }
+    }
+}
+
+/// Type of the plugin. Defaults to OncePerPane.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum PluginType {
+    // TODO: A plugin with output thats cloned across every pane in a tab, or across the entire
```

Comment:
Agreed! I think this might be helpful as another type of `Pane` as opposed to just a plugin type :) Happy to chat about it down the line!

Reply:
Yup! Someone in the Discord was doing a pomodoro timer in the Discord the other day, and they we're asking for what was essentially a "static" plugin pane, or a plugin that runs once and mirrors itself to every pane. I'll open an issue to discuss it after this gets merged.